checks/phpcs.php

```diff
@@ -110,16 +110,42 @@ protected function phpcs_result_to_warnings( $result ) {
 				}
 
 				$source_code = esc_html( trim( file( $this->path . '/' . $filename )[ $message['line'] - 1 ] ) );
+				$edit_link   = '';
+
+				if ( ! defined( 'DISALLOW_FILE_EDIT' ) || ! DISALLOW_FILE_EDIT ) {
+					$plugin_data = get_plugins( '/' . $this->slug );
+
+					$edit_link = sprintf(
+						'<a href="%1$s" title="%2$s" aria-label="%2$s" target="_blank">%3$s</a>',
+						esc_url(
+							add_query_arg(
+								[
+									'plugin' => rawurlencode( $this->slug . '/' . array_key_first( $plugin_data ) ),
+									'file'   => rawurlencode( $this->slug . '/' . $filename ),
+									'line'   => rawurlencode( $message['line'] ),
+								],
+								admin_url( 'plugin-editor.php' )
+							)
+						),
+						sprintf(
+							/* translators: %1$s is the path to a plugin file. */
+							esc_attr__( 'View %1$s in the plugin file editor.', 'plugin-check' ),
+							$this->slug . '/' . $filename
+						),
+						esc_html__( 'View in code editor', 'plugin-check' )
+					);
+				}
```

Comment:
Even if `DISALLOW_FILE_EDIT` is true, the file editor can still be opened so the user can at least see the code in question even if they can't edit it. Upon attempting to save, they get an error:

![image](https://github.com/10up/plugin-check/assets/134745/5f603a49-52e5-4207-a967-1bf7a1589a85)

Also, what about providing a filter to allow users to open their IDE of choice? We did this in the AMP plugin via an `amp_validation_error_source_file_editor_url_template` filter:

https://github.com/ampproject/amp-wp/blob/956c8d298764e4b22b936347d7c8b591b2fadfc2/includes/validation/class-amp-validation-error-taxonomy.php#L2356-L2473